presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -748,6 +748,34 @@ public void testGreatCircleDistance()
         assertInvalidFunction("great_circle_distance(36.12, -86.67, 33.94, infinity())", "Longitude must be between -180 and 180");
     }
 
+    @Test
+    public void testSTInteriorRings()
+    {
+        assertInvalidFunction("transform(ST_InteriorRings(ST_GeometryFromText('POINT (2 3)')), x -> ST_ASText(x))",
+                "ST_InteriorRings only applies to POLYGON. Input type is: POINT");
+        assertInvalidFunction("transform(ST_InteriorRings(ST_GeometryFromText('LINESTRING EMPTY')), x -> ST_ASText(x))",
+                "ST_InteriorRings only applies to POLYGON. Input type is: LINE_STRING");
+        assertInvalidFunction("transform(ST_InteriorRings(ST_GeometryFromText('MULTIPOINT (30 20, 60 70)')), x -> ST_ASText(x))",
+                "ST_InteriorRings only applies to POLYGON. Input type is: MULTI_POINT");
+        assertInvalidFunction("transform(ST_InteriorRings(ST_GeometryFromText('MULTILINESTRING ((1 10, 100 1000), (2 2, 1 0, 5 6))')), x -> ST_ASText(x))",
+                "ST_InteriorRings only applies to POLYGON. Input type is: MULTI_LINE_STRING");
+        assertInvalidFunction("transform(ST_InteriorRings(ST_GeometryFromText('MULTIPOLYGON (((1 1, 1 3, 3 3, 3 1)), ((0 0, 0 2, 2 2, 2 0)))')), x -> ST_ASText(x))",
+                "ST_InteriorRings only applies to POLYGON. Input type is: MULTI_POLYGON");
+        assertInvalidFunction("transform(ST_InteriorRings(ST_GeometryFromText('GEOMETRYCOLLECTION (POINT (1 1), POINT (2 3), LINESTRING (5 8, 13 21))')), x -> ST_ASText(x))",
+                "ST_InteriorRings only applies to POLYGON. Input type is: GEOMETRY_COLLECTION");
+
+        assertSTInteriorRings("POLYGON EMPTY", null);
+        assertSTInteriorRings("POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))", ImmutableList.of());
+        assertSTInteriorRings("POLYGON ((0 0, 0 3, 3 3, 3 0, 0 0), (1 1, 1 2, 2 2, 2 1, 1 1))", ImmutableList.of("LINESTRING (1 1, 1 2, 2 2, 2 1, 1 1)"));
+        assertSTInteriorRings("POLYGON ((0 0, 0 5, 5 5, 5 0, 0 0), (1 1, 1 2, 2 2, 2 1, 1 1), (3 3, 3 4, 4 4, 4 3, 3 3))",
+                ImmutableList.of("LINESTRING (1 1, 1 2, 2 2, 2 1, 1 1)", "LINESTRING (3 3, 3 4, 4 4, 4 3, 3 3)"));
+    }
+
+    private void assertSTInteriorRings(String wkt, ImmutableList<String> expected)
```

Comment:
for readability, let's drop `ST` from the method name and use varargs:

`private void assertInteriorRings(String wkt, String... expected)`